### PR TITLE
Switch to GSON. Add POST method for comments.

### PR DESCRIPTION
**Pull request by tin0d0**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -24,31 +25,39 @@
...
+    response.sendRedirect("/index.html");
+  }
+
+  /**
+   * Converts a ServerStats instance into a JSON string using the Gson library. Note: We first added
+   * the Gson library dependency to pom.xml.
+   */
+  private String convertToJsonUsingGson(ArrayList<String> comments) {
+    Gson gson = new Gson();
+    String json = gson.toJson(comments);
```
Comment (jileiwang-google) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Nit: Is the gson object reusable? If so, you can also create a Gson as class member, then use it directly in line 35.

**Review by jileiwang-google — APPROVED** [review]
LGTM

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (tin0d0) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Awesome, I just changed it to that! Thank you!

File: portfolio/src/main/java/com/google/sps/Data/Comment.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
+// Copyright 2019 Google LLC
```
Comment (HidyHan) [inline portfolio/src/main/java/com/google/sps/Data/Comment.java:1]: You are gonna Google copy right for this file? :p (no action required)

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -27,29 +36,45 @@
...
-    json += "\"" + array.get(1) + "\"";
-    json += ", ";
-    json += array.get(2);
-    return json;
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      
+    String userCommentString = request.getParameter("user-comment");
+    String userNameString = request.getParameter("user-name");
+    long timestamp = System.currentTimeMillis();
```
Comment (HidyHan) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:66]: Doesn't matter here, but if you were gonna execute two RPC calls before retrieving the timestamp, better do it the first thing.

**Review by HidyHan — APPROVED** [review]

**Pull request merged**